### PR TITLE
New gamemode experiment

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

https://youtu.be/2PjCClrO5SA

Adds a unfinalized new "Garrison Defence" gamemode
Summed up:
- Both teams get a "base": marines a computer-thing, xenos a more powerful silo
- When this structure is destroyed you lose
- When no members of your team are alive you lose
- The bases start shielded and you have to destroy large turrets to destroy the shield
- No shipside combat, though ship still exists
- Both teams should start separated by a shutter so they can set up

TLDR -> kill turrets -> kill base -> win

Todo
- [ ] Mapping
- [ ] Just realized probably should announce base becoming vulnerable
- [ ] probably forgot something

## Why It's Good For The Game

Shipside combat is rather hated

Additionally there's a rather persistent issue where xenos are melee (hence need to be aggressive) but defend by default, and marines get defensive structures that are either useless for 90% of the round (like now) or FOBcamp hell, this hopefully alleviates some of those issues by allowing both teams to attack/defend off the bat.

Also adds proper round end objectives that ideally would encourage teamwork but I'm a little doubtful there.

Otherwise we'll see what happens during TM

## Changelog
:cl:
add: Added new Garrison Defence gamemode
/:cl:

### CODE REVIEW

**Comment by Al-1ce** [comment]
Marines generally have more avenues/pressure to push with than xenos, so I'm curious to see how this will work out.

>90% of the round (like now) or FOBcamp hell

If either base is surrounded by turrets, _especially_ on the marine side, wouldn't this just exacerbate that problem?

**Comment by Losenis** [comment]
>No shipside
>No retarded fob building that half the time is either shit or nonexistent
>Actual solid objectives to get ungas to go there rather than dumb out spread all over the place with some going disk, others kill beno, others just going to a different disk than the first, others going gens, etc

Incredibly based, hyped for this

**Comment by zee11zee** [comment]
Leauge update??!! sounds fun, cant wait.

**Comment by FIXTHERADBOXPLEASE** [comment]
The PR that saved TGMC.

**Comment by iforgotmeword** [comment]
Would they have to break all or just some? Because that would be a real hell think a mega FOB with all of the marines holding it. And xenos can’t do anything to stop the marines firing fire support from the LZ

also would you be able to repair the turret or even replace it?

File: code/datums/gamemodes/garrison.dm [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	UnregisterSignal(marine_base, COMSIG_PARENT_QDELETING)
+	marine_base = null
+	round_finished = MODE_INFESTATION_X_MAJOR
+	message_admins("Round finished: [MODE_INFESTATION_X_MAJOR]")
+
+/// Marine base deleted without deconstruct being called aka admin or bug deleted it
+/datum/game_mode/infestation/garrison/proc/marine_base_deleted(datum/source)
+	SIGNAL_HANDLER
+	marine_base = null
+	message_admins("Marine base deleted unexpectedly. Please end the round manually or replace with a new xeno base using VV.")
```
Comment (XSlayer300) [inline code/datums/gamemodes/garrison.dm:96]:
```suggestion
	message_admins("Marine base deleted unexpectedly. Please end the round manually or replace with a new marine base using VV.")
```

File: code/modules/xenomorph/xeno_structures.dm [hunk 2]
```diff
@@ -850,6 +851,13 @@ TUNNEL
 	SSminimaps.remove_marker(src)
 	SSminimaps.add_marker(src, z, MINIMAP_FLAG_XENO, "silo[warning ? "_warn" : "_passive"]")
 
+/obj/structure/xeno/silo/core
+	name = "resin silo repository"
+	desc = "A slimy, rank surface blob of green...stuff restrained with resin. Somethings moving in there."
```
Comment (XSlayer300) [inline code/modules/xenomorph/xeno_structures.dm:856]:
```suggestion
	desc = "A slimy, rank surface blob of green... stuff restrained with resin. Something is moving in there, probably best for humans to stay away from it."
```

**Comment by Strayed-Raven** [comment]
Wonder if this will subsume Nuke and then balance just gets worked around for this mode instead.

**Review by Kuro020 — APPROVED** [review]
tivi doing work? not my tivi

**Comment by chizzy376** [comment]
Holy shit TGMC MOBA

**Comment by chizzy376** [comment]
Jokes aside, think this could also be possible for Human vs Human?

**Comment by MereWalker** [comment]
I've been asking for this for years! Now it comes to fruition!

**Comment by Autisem** [comment]
To people saying it will make the mega FOB problem worse, maby to some extend but I think marines will have difficult time defending a spot they cannot retreat from especialy of you have a good mix of certain xenos

**Pull request closed without merging**